ApollosProject/apollos-prototype
# Notifications Soft Ask

## DESCRIPTION

### What does this PR do, or why is it needed?

Adds a soft ask for enabling notifications. This should fix an issue where after rejecting notifications, the "enable notifications" button did nothing and could be confusing.

The soft ask checks that a user has been prompted, then if so, uses an alert to ask the user to enable notifications. If they accept, then the user will be routed to the app settings to enable notifications.

This soft ask will ONLY occur if the user has been previously hard asked AND they press the "Enable Notifications" button in onboarding.

### What design trade-offs/decisions were made?

### How do I test this PR?

1. In onboarding tap Enable Notifications and reject the hard ask
2. Log out
3. In onboarding tap Enable notifications; you should receive a soft ask
4. Reject it; nothing should happen
5. In onboarding tap Enable notifications; you should receive a soft ask
6. Accept it and turn your notifications on
7.  Return to the app; the state should be updated to reflect the change

### What automated tests did you write?

## SCREENSHOTS

<img width="570" alt="Screen Shot 2019-11-20 at 4 05 25 PM" src="https://user-images.githubusercontent.com/29125070/69278084-95491b00-0baf-11ea-9e9c-53327e52d408.png">

## TODO:


## REVIEW:

**Manual QA**


**Code Review: Questions to consider**


> The purpose of PR Review is to _improve the quality of the software._
